emperor/support_files/js/view-controller.js

```diff
@@ -488,16 +516,59 @@ define([
    *
    */
   EmperorAttributeABC.prototype.fromJSON = function(json) {
-    this.$select.val(json.category);
-    this.$select.trigger('chosen:updated');
+    this.setMetadataField(json.category);
 
     // fetch and set the SlickGrid-formatted data
-    var k = this.getActiveDecompViewKey();
-    var data = this.decompViewDict[k].setCategory(
+    var data = this.getView().setCategory(
       json.data, this.setPlottableAttributes, json.category);
     this.setSlickGridDataset(data);
     // set all to needsUpdate
-    this.decompViewDict[k].needsUpdate = true;
+    this.getView().needsUpdate = true;
+  };
+
+  /**
+   *
+   * Update the metadata selection menu.
+   *
+   * Performs some additional logic to avoid duplicating decomposition names.
+   *
+   * Note that decompositions won't be updated if they have the same name and
+   * same metadata headers, if the only things changing are coordinates, or
+   * metadata values, the changes should be performed directly on the objects
+   * themselves.
+   *
+   */
+  EmperorAttributeABC.prototype.refreshMetadata = function() {
```

Comment:
Just to clarify, this refreshing all of the metadata associated with the current view? 

Remember, every decomposition view will have their own set of metadata.  

Samples -> sample metadata (ex ph, location)
Features -> feature metadata (ex taxonomy )

If not,  then this will need to be clarified in the docs, along with a todo/issue.